Modules/Messages/src/Messages.Domain/Entities/Message.cs

```diff
@@ -86,11 +86,13 @@ private void AnonymizeRecipient(IdentityAddress participantAddress, IdentityAddr
     {
         var recipient = Recipients.FirstOrDefault(r => r.Address == participantAddress);
         recipient?.UpdateAddress(anonymizedIdentityAddress);
+        RaiseDomainEvent(new MessageOrphanedDomainEvent(this));
```

Comment:
Wtf. Why do you ALWAYS raise the `MessageOrphanedDomainEvent`, no matter if there are un-anonymized participants left?

Reply:
I agree it's better to perform the check before raising the `MessageOrphanedDomainEvent`. Changes made.

Reply:
Well, as soon as the event handler deletes the orphaned messages without any additional check it has to rely on the event to only be raised if the message is actually orphaned. With the current implementation the messages would be deleted as soon as the first participant is anonymized.

Reply:
In this implementation my logic for checking weather the message is orphaned or not is done in the `Delete` method in `MessageRepository` with the help of an expression,

```csharp
    public async Task Delete(Expression<Func<Message, bool>> filter, CancellationToken cancellationToken)
    {
        await _messages.Where(filter).ExecuteDeleteAsync(cancellationToken);
    }
 ```

the expression:

```csharp
    public static Expression<Func<Message, bool>> IsMessageOrphaned(string didDomainName)
    {
        var anonymizedAddress = IdentityAddress.GetAnonymized(didDomainName);

        return i =>
            i.CreatedBy == anonymizedAddress &&
            i.Recipients.All(r => r.Address == anonymizedAddress);
    }
 ```
 
 I raised every time the event when someone is anonymized (which now I understand it's not a good solution). Also I manually tested this. First I started a deletion process for identity A, run the job and the message wasn't deleted. Later on I started a deletion process for identity B and then the message was indeed deleted. As I mentioned in some of the previous comments, I tested everything manually otherwise I wouldn't consider the PR ready for review.